lib/cog/commands/filter.ex

```diff
@@ -26,29 +35,55 @@ defmodule Cog.Commands.Filter do
     {:reply, req.reply_to, result, state}
   end
 
-  defp maybe_filter(item, %{"field" => field, "matches" => matches}) do
-    match = with {:ok, field} <- Access.fetch(item, field),
-      regex = compile_regex(matches),
-      field_string = to_string(field),
-      do: String.match?(field_string, regex)
-
-    case match do
-      true ->
-        item
-      _ ->
-        nil
-    end
+  defp maybe_filter(item, %{"path" => path, "matches" => matches}) do
+    build_path(String.split(path, "."), [], [])
+    |> fetch(item, matches)
   end
-  defp maybe_filter(item, %{"field" => field}) do
-    case Access.fetch(item, field) do
+  defp maybe_filter(item, %{"path" => path}) do
+    build_path(String.split(path, "."), [], [])
+    |> fetch(item)
+  end
+  defp maybe_filter(_item, %{"matches" => _matches}),
+    do: %{error: "Please include a `path` to match with"}
+  defp maybe_filter(item, _),
+    do: item
+
+  defp fetch([], item), do: item
+  defp fetch([key|path], item) do
+    result = case Access.fetch(item, key) do
       {:ok, _} ->
-        item
+        fetch_path(true, path, item, key)
       :error ->
         nil
     end
+    fetch(path, result)
   end
-  defp maybe_filter(item, _),
-    do: item
+
+  defp fetch([], item, matches) do
+    match_str = Regex.compile!(matches)
+    match = Regex.match?(match_str, to_string(item))
+    fetch_match(match, item)
+  end
+  defp fetch(_, nil, _), do: nil
+  defp fetch([key|path], item, matches) do
+    result = with {:ok, value} <- Access.fetch(item, key),
+      regex = compile_regex(matches),
```

Comment:
I recommend checking out the implementation of the `rules` and `permissions` command for a useful processing pattern that makes managing errors easier.